src/Tools/GetDocumentInsider/tests/GetDocumentTests.cs

```diff
@@ -144,4 +144,86 @@ public void GetDocument_WithInvalidDocumentName_Errors()
         Assert.False(File.Exists(Path.Combine(outputPath.FullName, "Sample_internal.json")));
         Assert.False(File.Exists(Path.Combine(outputPath.FullName, "Sample_invalid.json")));
     }
+
+    [Theory]
+    [InlineData("customFileName")]
+    [InlineData("custom-File_Name")]
+    [InlineData("custom_File-Name")]
+    [InlineData("custom1File2Name")]
+    public void GetDocument_WithValidFileName_Works(string fileName)
+    {
+        // Arrange
+        var outputPath = Directory.CreateDirectory(Path.Combine(Path.GetTempPath(), Path.GetRandomFileName()));
+        var app = new Program(_console);
+
+        // Act
+        app.Run([
+            "--assembly", _testAppAssembly,
+            "--project", _testAppProject,
+            "--framework", _testAppFrameworkMoniker,
+            "--tools-directory", _toolsDirectory,
+            "--output", outputPath.FullName,
+            "--file-list", Path.Combine(outputPath.FullName, "file-list.cache"),
+            "--file-name-override", fileName
```

Comment:
The tests are failing because this needs to be updated to the new command line option name.